### PR TITLE
[Mac] Move appearance setting calls outside ViewDidChangeEffectiveApp…

### LINKED ISSUE #587: Move ViewDidChangeEffectiveAppearance calls
We need to move our effective appearance calls to a second method that we can call from the ctor and an override.

### PR DESCRIPTION
**Pull request by CartBlanche**
…erance, so it works on earlier MacOS versions.

Fixes #587 .

### CODE REVIEW

**Comment by CartBlanche** [comment]
Seems this VS4Mac PR https://github.com/xamarin/monodevelop/pull/552 may need this change before their PR can go through.

File: Xamarin.PropertyEditing.Mac/Controls/Custom/ColorEditorView.cs [hunk 1]
```diff
@@ -43,6 +43,10 @@ public virtual void OnViewModelChanged (TViewModel oldModel)
 		public virtual void OnPropertyChanged (object sender, PropertyChangedEventArgs e)
 		{
 		}
+
+		protected virtual void AppearanceChanged ()
```
Comment (alanmcgovern) [inline Xamarin.PropertyEditing.Mac/Controls/Custom/ColorEditorView.cs]: Can we add this declaration here:

```
public sealed override void ViewDidChangeChangeEffectiveAppearance ()
{
    base.ViewDidChangeEffectiveAppearance ();
    AppearanceChanged ();
}
```
This gives us two main advantage:
1) We are guaranteed that ViewDidChangeEffectiveAppearance is overridden once for all subclasses, so AppearanceChanged will only be invoked once every time ViewDidChangeEffectiveAppearance is invoked. Without this, if there are multiple subclasses then each of them may override ViewDidChangeEffectiveAppearance and invoke AppearanceChanged, which means we'll get multiple invocations of AppearanceChanged.
2) It ensures that no-one accidentally puts code in ViewDidChangeEffectiveAppearance in other subclasses.

On a related note, we should probably seal every override of this method for the reasons :)

File: Xamarin.PropertyEditing.Mac/PropertyEditorPanel.cs [hunk 2]
```diff
@@ -197,16 +197,21 @@ private void Initialize ()
...
-			ViewDidChangeEffectiveAppearance ();
+			UpdateResourceProvider ();
 		}
 
-		private void UpdateResourceProvider()
+		public override void ViewDidChangeEffectiveAppearance ()
+		{
+			base.ViewDidChangeEffectiveAppearance ();
+
+			UpdateResourceProvider ();
```
Comment (alanmcgovern) [inline Xamarin.PropertyEditing.Mac/PropertyEditorPanel.cs]: Should this method be renamed `AppearanceChanged` for consistency with every other implementation?

File: Xamarin.PropertyEditing.Mac/PropertyList.cs [hunk 3]
```diff
@@ -91,10 +91,9 @@ public IHostResourceProvider HostResourceProvider
 
 		public override void ViewDidChangeEffectiveAppearance ()
 		{
-			if (this.propertyTable == null)
-				return;
+			base.ViewDidChangeEffectiveAppearance ();
 
-			this.propertyTable.BackgroundColor = this.hostResources.GetNamedColor (NamedResources.PadBackgroundColor);
+			UpdateResourceProvider ();
```
Comment (alanmcgovern) [inline Xamarin.PropertyEditing.Mac/PropertyList.cs]: Should this method be renamed `AppearanceChanged` for consistency with every other implementation?

File: Xamarin.PropertyEditing.Mac/PropertyEditorPanel.cs (hunk 2 above)
Comment (CartBlanche) [reply Xamarin.PropertyEditing.Mac/PropertyEditorPanel.cs]: @alanmcgovern I did consider it, but currently it specifically only updates the respective ResourceProvider and not any other Appearance values for specific control. If you really want it it changed, I can do that.

File: Xamarin.PropertyEditing.Mac/Controls/Custom/ColorEditorView.cs (hunk 1 above)
Comment (CartBlanche) [reply Xamarin.PropertyEditing.Mac/Controls/Custom/ColorEditorView.cs]: @alanmcgovern When you say *here*, I assume you actually mean in the actual base class. Just asking to be 100% clear.

Comment (CartBlanche) [reply Xamarin.PropertyEditing.Mac/Controls/Custom/ColorEditorView.cs]: Hmm Nm, This makes sense, but highlights I need to do this in the *other* base class.

File: Xamarin.PropertyEditing.Mac/PropertyEditorPanel.cs (hunk 2 above)
Comment (alanmcgovern) [reply Xamarin.PropertyEditing.Mac/PropertyEditorPanel.cs]: I suppose i'm just looking for a consistent set of `sealed override ViewDidChangeEffectiveAppearance` and `void UpdateAppearance` methods.

Let's put in a pin in this for now as i just did a quick code review of the rest of the codebase and have some concerns. I'll file an issue with my thoughts!

File: Xamarin.PropertyEditing.Mac/Controls/Custom/ColorEditorView.cs (hunk 1 above)
Comment (CartBlanche) [reply Xamarin.PropertyEditing.Mac/Controls/Custom/ColorEditorView.cs]: Done!

Comment (alanmcgovern) [reply Xamarin.PropertyEditing.Mac/Controls/Custom/ColorEditorView.cs]: Ah yeah, i meant in this base class and all other places where the method is overridden :)

**Comment by alanmcgovern** [comment]
🎉

**Pull request merged**